### PR TITLE
VoxelFilter for TimedPointCloud, Filter method.

### PR DESCRIPTION
**Pull request by gaschler**
This makes the VoxelFilter ready for TimedPointCloud and
per-point unwarping.

### CODE REVIEW

File: cartographer/sensor/voxel_filter.cc [hunk 1]
```diff
@@ -77,25 +78,32 @@ PointCloud AdaptivelyVoxelFiltered(
...
-  VoxelFilter voxel_filter(size);
-  voxel_filter.InsertPointCloud(point_cloud);
-  return voxel_filter.point_cloud();
-}
-
 VoxelFilter::VoxelFilter(const float size) : voxels_(size) {}
 
-void VoxelFilter::InsertPointCloud(const PointCloud& point_cloud) {
-  for (const Eigen::Vector3f& point : point_cloud) {
+PointCloud VoxelFilter::Filter(const PointCloud& point_cloud) {
```
Comment (cschuet) [inline cartographer/sensor/voxel_filter.cc]: template implementation of this? You can also call .head<3> on Vector3 at no cost.

**Comment by gaschler** [comment]
Done with the corrections, PTAL

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 2.

**Pull request merged**